egs/wsj/s5/steps/libs/nnet3/xconfig/parser.py

```diff
@@ -49,11 +49,63 @@ def xconfig_line_to_object(config_line, prev_layers = None):
               "*** {0}".format(config_line), file=sys.stderr)
         raise e
 
+# This function reads existing model file with nnet3 format and returns it as
+# list of layers with name and dimension to be used as auxilary information
+# to generate xconfig.
+def read_model(model_filename):
```

Comment:
there is no documentation of what the return format of this function is, and the name is too non-specific (e.g. get_model_component_info).  Don't use the word 'layer' if what you really mean is component, they are different things.  And grep for '^component-node', not -node.  You could otherwise get problems in future if something that looks like that appears in the file.